### ISSUE TITLE
ODataConventionModelBuilder.EnableLowerCamelCase() breaks navigation properties

### ISSUE THREAD

**Issue by BlaineM-SeriouslyRAD** [OData/AspNetCoreOData#391]
Hi,

Following on from my comment on issue #251 (https://github.com/OData/AspNetCoreOData/issues/251#issuecomment-937673732), I attach a sample demonstrating the problem (Microsoft.AspNetCore.OData v8.0.4).

Initially, the sample code has ODataConventionModelBuilder.EnableLowerCamelCase() commented out.

If you run the sample in IIS Express and navigate to https://localhost:44310/$odata, the list of mapped endpoints is as listed below.

However, uncommenting EnableLowerCamelCase() (Startup.cs, Line 67) and running again results in the loss of 4 endpoints (_**bold + italic**_ below).

- odata/ProductGroups
- odata/ProductGroups/$count
- odata/ProductGroups({key})
- odata/ProductGroups/{key}
- odata/ProductGroups({key})/ODataSample.Models.ProductGroup
- odata/ProductGroups/{key}/ODataSample.Models.ProductGroup
- _**odata/ProductGroups({key})/Products**_
- _**odata/ProductGroups/{key}/Products**_
- _**odata/ProductGroups({key})/Products/$count**_
- _**odata/ProductGroups/{key}/Products/$count**_
- odata/$metadata
- odata

This worked fine in v7.5.11.

[ODataSample.zip](https://github.com/OData/AspNetCoreOData/files/7591326/ODataSample.zip)

**Comment by xuzhg** [comment]
@BlaineM-SeriouslyRAD [#251 (comment)](https://github.com/OData/AspNetCoreOData/issues/251#issuecomment-937673732) is fixed the query option parts when doing request process.

For the convention routing table building, it's still using the "case-sensitive".
see https://github.com/OData/AspNetCoreOData/blob/master/src/Microsoft.AspNetCore.OData/Routing/Conventions/PropertyRoutingConvention.cs#L109.

Basically, the OData conventional routing is based on:

1) The controller name (Case-sensitive)
2) the action name (case-sensitive).

I am thinking about to enable the case-insensitive.

**Comment by xuzhg** [comment]
4f195c8835a7931eae111c0c5b272f831aa94521

**Comment by BlaineM-SeriouslyRAD** [comment]
Thanks very much @xuzhg. I look forward to the release 😀

**Comment by BlaineM-SeriouslyRAD** [comment]
Hi @xuzhg,

Many thanks for including the fix for this in v8.0.5.
I've removed our patch, installed v8.0.5 and set EnablePropertyNameCaseInsensitive to true on the RouteOptions.

All appears to be working as expected 😊

Blaine
